### DISCUSSION TITLE
Unable to get a response from the time-consuming API service

### DISCUSSION THREAD

**Discussion by kelvennn** [fastapi/fastapi#12900] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import os
import asyncio
import datetime
from functools import partial
import json
from typing import List, Optional
import pandas as pd
from fastapi import FastAPI
import uvicorn
import requests
from pydantic import BaseModel
from fastapi.responses import JSONResponse, Response, StreamingResponse
from multiprocessing import Process, Queue
import multiprocessing
import time
from tqdm import tqdm

class CustomRequest(BaseModel):
    wait_time: int = 1

app = FastAPI(docs_url='/')

@app.post('/test')
async def test(request:CustomRequest):
    response = {
        'code': 0,
        'msg': 'success',
        'result': []
    }
    for i in tqdm(range(request.wait_time)):
        time.sleep(1)
    return JSONResponse(response)

def serve(server_name: str = '0.0.0.0',
          server_port: int = 9254):
    uvicorn.run(app=app,
                host=server_name,
                port=server_port,
                log_level='info')

if __name__ == '__main__':
    serve()
```

### Description

I have a time-consuming API service deployed on a remote server, I call the API from the local program, when the calculation time 
 exceeds 40min, my local program won't get any response and finally occur timeout error, despite the server display:
INFO: ****:45748 - "POST /test HTTP/1.1" 200 OK

### Operating System

Linux

### FastAPI Version

0.115.2

### Pydantic Version

2.9.2

### Python Version

python 3.10.14

**Comment by luzzodev** [comment, marked as answer]
Hello @kelvennn, supposing your example is the remote API,

if a client disconnection occur ( your local program), Fastapi/Starlette will continue to handle the request as the client in still connected.
You can check if connection is active by quering the request.

In your case

```python
from fastapi import Request
...
@app.post('/test')
async def test(request: Request, custom_request:CustomRequest):
  ...
  # check for disconnection before return response
  # use: await request.is_disconnected()
```
